io/plugins/eu.esdihumboldt.hale.io.jdbc/src/eu/esdihumboldt/hale/io/jdbc/JDBCTableCollection.java

```diff
@@ -242,8 +242,13 @@ public JDBCTableCollection(TypeDefinition type, URI jdbcURI, String user, String
 
 			// support project variables
 			query = JDBCUtil.replaceVariables(query, services);
-
-			this.countQuery = null;
+			
+			// this.countQuery = null; 
+			// countQuery = null caused to return UNKNOWN_SIZE in size() with causes isEmpty() 
+			// to return false and that causes problems on iterating with MultiInstanceCollection 
+			this.countQuery = "SELECT COUNT(*) FROM (\n" + query + "\n) tmp";
+			// note 1: this sub query is not supported in all SQL dialects
```

Comment:
I am not sure if subquery is part of standard SQL syntax. However accoring to a quick research all major databases seem to support.